src/ai/backend/manager/registry.py

```diff
@@ -1273,10 +1272,54 @@ async def _finialize_running() -> None:
                             'stdin_port': created_info['stdin_port'],
                             'stdout_port': created_info['stdout_port'],
                             'service_ports': service_ports,
-                        })
-                        .where(kernels.c.id == created_info['id']))
-                    await conn.execute(query)
-
+                        }
+                        actual_allocs: MutableMapping[str, str] = {}
+                        for alloc_map in created_info['resource_spec']['allocations'].values():
+                            for slot_name, allocations in alloc_map.items():
+                                total_allocs: List[Decimal] = []
+                                for allocation in allocations.values():
+                                    if BinarySize.suffix_map.get(allocation[-1].lower()) is not None:
+                                        total_allocs.append(Decimal(BinarySize.from_str(allocation)))
+                                    else:
+                                        total_allocs.append(Decimal(allocation))
+                                actual_allocs[slot_name] = str(sum(total_allocs))
+
+                        values['occupied_slots'] = actual_allocs
+                        query = (
```

Comment:
Let's name the variables that points sqlalchemy queries differently, e.g., `select_query` and `update_query`, etc. (or maybe `read_query`, `info_query`, etc.).
This is to prepare adoption of sqlalchemy typing in the future and avoid conflicts of different types of the same-named variables.